src/core/src/local_loggers/CompositeLogger.py

```diff
@@ -34,9 +38,17 @@ def __init__(self, file_logger=None, current_env=None):
     @staticmethod
     def log(message):
         """log output"""
-        message = CompositeLogger.__remove_substring_from_message(message, Constants.ERROR_ADDED_TO_STATUS)
-        for line in message.splitlines():  # allows the extended file logger to strip unnecessary white space
-            print(line)
+        try:
+            message = CompositeLogger.__remove_substring_from_message(message, Constants.ERROR_ADDED_TO_STATUS)
+            for line in message.splitlines():  # allows the extended file logger to strip unnecessary white space
+                sys.stdout.write(line)
+                sys.stdout.flush()
```

Comment:
Please note that print implicitly includes a new line feed at the end of the line, which sys.stdout.write() doesn't (so you'd have to add one). Also is this a confirmed or a hypothesis? I don't expect this to work because:
 - sys.stdout is not really sys.stdout here. It's StdOutFileMirror, which deliberately does nothing in a flush scenario.
 - If sys.stdout was really sys.stdout with a functional flush(), it would only expedite the failure. As noted below, we cannot ignore the error if we don't have a path to recovery.

The exception handler here would not catch anything except a failure to write to the terminal (it's good to understand if this is the actual cause -- simply comment out line 37 in StdOutFileMirror for this and can be the entirety of the fix on a trial basis). 

All errors occurring outside of that line including broken pipe error will be caught in two levels of exception handling in FileLogger and CompositeLogger. And failures will be written to a file with the same name as the log file but with an additional '.failure' extension.

Reply:
**Copying from the other PR since it pertains to this actually:**
I’m pretty much convinced the broken pipe error happens because the extension handler captures the stdout and stderr and later goes away while Core continues to write to it.

See here for someone complaining about this:
https://stackoverflow.com/questions/4106565/newbie-python-subprocess-write-error-broken-pipe

This was never a problem in V1 where what is now Core had the exact same behavior (relating to output/errors) but had a long running equivalent process capture that never went away and constantly kept reading from the IO buffers.

Basically this issue needs to be addressed first in the extension handler and to a lesser degree in Core.

We shouldn’t get a sigpipe if we redirect stdout and stderr to devnull when the Core process is created:

with open(os.devnull, 'r+b', 0) as DEVNULL:
    process = Popen(command, stdin=DEVNULL, stdout=DEVNULL, stderr=STDOUT, close_fds=True)

What this does mean for code in lines 86 and 87 is that IF the process doesn’t start correctly, we can run it again in the mode we run it right now and capture what the failure is.

For Core itself, this means that we can doubly guard against this by comment out line 37 in StdOutFileMirror or try to put exception handling around it (line 37) and ignore any exceptions from writing to the “terminal” which is captured by the dead extension handler process.

Reply:
modified